nkms/client.py

```diff
@@ -67,6 +71,43 @@ def _split_path(self, path):
         dirs = path.split(b'/')
         return [b'/'.join(dirs[:i + 1]) for i in range(len(dirs))]
 
+    def _build_header(self, enc_keys, version=100):
+        """
+        Creates a NuCypher header for the encrypted file.
+
+        :param enc_keys: List of encrypted keys in bytes
+        :param version: Version number of Cryptographic API (default: 0.1.0.0)
+
+        :return: Complete header msgpack encoded and length of raw header
+        :rtype: Tuple of the header and the header length e.g: (<header>, 1200)
+        """
+        if version < 1000:
+            vers_bytes = version.to_bytes(4, byteorder='big')
+            num_keys_bytes = len(enc_keys).to_bytes(4, byteorder='big')
+            keys = b''.join(enc_keys)
+            header = msgpack.dumps(vers_bytes + num_keys_bytes + keys)
```

Comment:
I think, the point of msgpack is to avoid these sorts of things. So, you can just say msgpack([version] + enc_keys), and it will care about everything.
Your way works also, but then there is no need to use msgpack